polygeist/lib/Conversion/PolygeistToLLVM/PolygeistToLLVM.cpp

```diff
@@ -938,15 +932,13 @@ struct ConvertPolygeistToLLVMPass
       LLVMTypeConverter converter(&getContext(), options, &dataLayoutAnalysis);
       RewritePatternSet patterns(&getContext());
 
-      if (useBarePtrCallConv) {
-        // Keep these at the top; these should be run before the rest of
-        // function conversion patterns.
-        populateReturnOpTypeConversionPattern(patterns, converter);
-        populateCallOpTypeConversionPattern(patterns, converter);
-        populateAnyFunctionOpInterfaceTypeConversionPattern(patterns,
-                                                            converter);
-      }
-      sycl::populateSYCLToLLVMConversionPatterns(converter, patterns);
+      // Keep these at the top; these should be run before the rest of
```

Comment:
Not related to this PR. Can we use benefit to ensure these patterns run before rest of the conversion patterns?

Reply:
I wish... `populateX` does not receive a `benefit` argument, so we have to rely on ordering here...